/****** Object:  Table [T_Step_Tools] ******/
/****** RowCount: 81 ******/
SET IDENTITY_INSERT [T_Step_Tools] ON
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (4,'MSMSSpectraPreprocessor','MSMS Spectrum Filter','Produce modified DTA by filtering existing DTA',0,1,1,0,250,'<section name="SpectraFilter" tool="MSMSSpectraPreprocessor" category="basic"><item key="FilterType" value="" /></section><section name="FilterOptions" tool="MSMSSpectraPreprocessor" category="advanced"><item key="FilterMode" value="3" /><item key="MinimumQualityScore" value="0.25" /><item key="GenerateFilterReport" value="True" /><item key="IncludeBPIAndNLStatsOnFilterReport" value="True" /><item key="OverwriteExistingFiles" value="True" /><item key="DiscardValidSpectra" value="False" /><item key="EvaluateSpectrumQualityOnly" value="False" /><item key="MSLevelFilter" value="0" /></section><section name="FilterMode3" tool="MSMSSpectraPreprocessor" category="advanced"><item key="BasePeakIntensityMinimum" value="1000" /><item key="MassToleranceHalfWidthMZ" value="0.7" /><item key="NLAbundanceThresholdFractionMax" value="0.5" /><item key="LimitToChargeSpecificIons" value="True" /><item key="ConsiderWaterLoss" value="True" /><item key="SpecificMZLosses" value="" /></section>','Y','','','MSMSSpectraPreprocessor',0.15,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (5,'Sequest','Peptide Search','Identify peptides by searching protein collection',0,0,1,0,100,'<section name="Search" tool="Sequest" category="basic"><item key="AJ_ParmFile" value="" /><item key="OrganismName" value="" /><item key="legacyFastaFileName" value="" /><item key="ProteinCollectionList" value="na" /><item key="ProteinOptions" value="seq_direction=forward" /></section>','N','\\gigasax\DMS_Parameter_Files\Sequest','','Sequest',120,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (6,'XTandem','Peptide Search','Identify peptides by searching protein collection',0,0,7,0,250,'<section name="Search" tool="Sequest" category="basic"><item key="ParamFileName" value="" /><item key="OrganismName" value="" /><item key="LegacyFastaFileName" value="" /><item key="ProteinCollectionList" value="na" /><item key="ProteinOptions" value="seq_direction=forward" /></section>','N','\\gigasax\DMS_Parameter_Files\XTandem','','XTandem',30,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (7,'Decon2LS','Deisotoping','Uses Decon2LS to deisotope spectra',0,0,1,0,250,null,'N','\\gigasax\DMS_Parameter_Files\Decon2LS','','Decon2LS',300,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (8,'Decon2LS_Agilent','Deisotoping','',0,0,1,0,250,null,'N','','','Decon2LS_Agilent',300,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (9,'ICR2LS','Deisotoping','Uses ICR-2LS to deisotope spectra for Bruker datasets',0,0,1,0,250,null,'N','\\gigasax\DMS_Parameter_Files\ICR2LS','','ICR2LS',120,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (10,'TIC_D2L','TIC','',0,0,1,0,250,null,'Y','\\gigasax\DMS_Parameter_Files\TIC_D2L','','TIC_D2L',3,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (11,'TIC_D2L_Agilent','TIC','',0,0,1,0,250,null,'Y','\\gigasax\DMS_Parameter_Files\TIC_D2L','','TIC_D2L_Agilent',3,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (12,'MASIC_Finnigan','MASIC','Run MASIC on Thermo .Raw files',0,0,1,0,250,null,'N','\\gigasax\DMS_Parameter_Files\MASIC','','MASIC_Finnigan',7,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (13,'MASIC_Agilent','MASIC','',0,0,1,0,250,null,'N','\\gigasax\DMS_Parameter_Files\MASIC','','MASIC_Agilent',7,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (14,'DataExtractor','Peptide Extraction','Creates loadable peptide ID file and first hits file; will also run AScore if we searched for Phospho STY',0,0,2,0,250,null,'Y','','CPU_Load set to 2 to prevent more than 3 copies of the Data Extractor running on one machine (we run Peptide Prophet for SEQUEST jobs, and it can use a lot of memory)','DataExtractor',0.75,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (15,'DTA_Gen','DTA Generator','Creates DTA files according to values in settings file',1,0,1,0,250,'<section name="DtaGenerator" tool="DTA_Gen" category="basic"><item key="DtaGenerator" value="extract_msn.exe" /></section><section name="Charges" tool="DTA_Gen" category="advanced"><item key="CreateDefaultCharges" value="True" /><item key="ExplicitChargeStart" value="0" /><item key="ExplicitChargeEnd" value="0" /></section><section name="ScanControl" tool="DTA_Gen" category="advanced"><item key="ScanStart" value="1" /><item key="ScanStop" value="999999" /><item key="MaxIntermediateScansWhenGrouping" value="1" /></section><section name="MWControl" tool="DTA_Gen" category="advanced"><item key="MWStart" value="200" /><item key="MWStop" value="5000" /></section><section name="IonCounts" tool="DTA_Gen" category="advanced"><item key="IonCount" value="35" /></section><section name="MassTol" tool="DTA_Gen" category="advanced"><item key="MassTol" value="3" /></section>','Y','\\gigasax\DMS_Parameter_Files\DTA_Gen','','DTA_Gen',15,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (16,'Results_Transfer','Results Folder Move','Moves results folder from Xfer to storage',0,0,1,0,250,null,'Y','','','Results_Transfer',0.001,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (17,'Inspect','Peptide Search','Identify peptides by searching protein collection',0,0,1,0,250,null,'N','\\gigasax\DMS_Parameter_Files\Inspect','','Inspect',30,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (18,'InspectDataExtractor','Peptide Extraction','Creates loadable peptide ID file and first hits file',0,0,1,0,250,null,'Y','','','InspectDataExtractor',0.75,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (19,'mzXMLSpecraPreprocessor','mzXML MSMS Spectrum Filter','',0,0,1,0,250,null,'Y','','','mzXMLSpecraPreprocessor',5,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (20,'XTandemDataExtractor','Peptide Extraction','Creates loadable peptide ID file and first hits file',0,0,1,0,250,null,'Y',null,'','XTandemDataExtractor',30,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (21,'Bogus','Peptide Search','Testing software, again',0,0,1,0,250,'<section name="Search" tool="Sequest" category="basic"><item key="AJ_ParmFile" value=""><field label="Parameter File" type="text" size="60" rules="trim|required|max_length[255]" maxlength="255"><chooser type="list-report.helper" Target="helper_aj_param_file/report" XRef="AJ_ToolName" Delimiter="," /></field></item><item key="OrganismName" value=""><field label="Organism" type="text" size="30" rules="trim|required|max_length[50]" maxlength="80"><chooser type="list-report.helper" Target="helper_organism/report" Delimiter="," /></field></item><item key="legacyFastaFileName" value=""><field label="Organism DB File" type="text" size="60" rules="trim|required|max_length[64]" maxlength="80" /></item><item key="ProteinCollectionList" value="na"><field label="Protein Collection List" type="area" rules="trim|max_length[4000]" rows="3" cols="60"><chooser type="list-report.helper" Target="helper_protein_collection/report" XRef="AJ_Organism" Delimiter="," /></field></item><item key="ProteinOptions" value="seq_direction=forward"><field label="Protein Options List" type="area" rules="trim|max_length[256]" rows="2" cols="60"><chooser type="picker.replace" PickListName="protOptSeqDirPickList" Delimiter="," /></field></item></section>','Y','','','Bogus',0.5,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (22,'DTA_Split','DTA Generator','Splits DTA files created by DTAGen',0,1,1,0,250,'<section name="ParallelInspect"><item key="SubDTAFileBaseName" Value="_Part@_dta.txt" /><item key="SubResultFileBaseName" Value="_Part@_inspect.txt" /></section>','Y','','','DTA_Split',5,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (23,'MSXML_Gen','XML File Generator','Creates mzXML or mzML files from .Raw files',1,0,1,0,250,'<section name="MSXMLGenerator"><item key="MSXMLGenerator" value="MSConvert.exe" /><item key="MSXMLOutputType" value="mzML" /><item key="CentroidMSXML" value="True" /><item key="CentroidPeakCountToRetain" value="-1" /><item key="RecalculatePrecursors" value="False" /><item key="RecalculatePrecursorsTool" value="RawConverter.exe" /></section>','Y','\\gigasax\DMS_Parameter_Files\MSXML_Gen','','MSXML_Gen',10,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (24,'InspectResultsAssembly','Peptide Extraction','Combines peptide search result files generated by running Inspect in parallel',0,1,1,0,250,'<section name="ParallelInspect"><item key="SubDTAFileBaseName" Value="_Part@_dta.txt" /><item key="SubResultFileBaseName" Value="_Part@_inspect.txt" /></section>','Y','','','InspectResultsAssembly',0.5,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (25,'MSClusterDTAtoDAT','MSMS Spectrum Filter','Produce an MSCluster-compatible DAT file from an existing DTA',0,0,1,0,250,null,'Y','\\gigasax\DMS_Parameter_Files\MSClusterDAT_Gen','','MSClusterDTAtoDAT',1,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (26,'DTA_Import','DTA Importer','Imports manually generated DTA files',0,0,1,0,250,'','Y','','','DTA_Import',5,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (27,'LTQ_FTPek','Deisotoping','Uses ICR-2LS to deisotope spectra for Finnigan datasets',0,0,1,0,250,null,'Y','\\gigasax\DMS_Parameter_Files\LTQ_FTPek','','LTQ_FTPek',120,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (28,'Decon2LS_V2','Deisotoping','Uses Decon2LS AutoProcessor to deisotope spectra (supports IMS data and RAPID)',0,0,1,0,250,null,'N','\\gigasax\DMS_Parameter_Files\Decon2LS','','Decon2LS_V2',480,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (29,'OMSSA','Peptide Search','Runs OMSSA',0,0,1,0,250,null,'Y','\\gigasax\DMS_Parameter_Files\OMSSA','','OMSSA',30,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (30,'LCMSFeatureFinder','Feature Detection','Groups deisotoped data from Decon2LS to form LC-MS Features',0,1,1,0,250,'<section name="LCMSFeatureFinder"><item key="LCMSFeatureFinderIniFile" value="FF_IMS_50ppm_2500Intensity_2009-12-21.ini" /></section>','N','\\gigasax\DMS_Parameter_Files\LCMSFeatureFinder','','LCMSFeatureFinder',3,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (31,'DTA_Refinery','DTA Generator','Reads DTA files (typically created by DeconMSn), then refines the parent mass for each spectrum using and X!Tandem search',1,0,4,0,250,'<section name="DtaGenerator" tool="DTA_Gen"><item key="DtaGenerator" value="DeconMSN.exe" /></section><section name="DTARefinery"><item key="DTARefineryXMLFile" value="DTARef_Lowess_NoMods.xml" /></section>','Y','\\gigasax\DMS_Parameter_Files\DTARefinery','','DTA_Refinery',35,1,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (32,'XTandem_HPC','Peptide Search','Runs XTandem on Chinook high-performance computer',0,0,0,0,250,null,'Y','\\gigasax\DMS_Parameter_Files\XTandem','','XTandem_HPC',30,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (33,'mzXML_Aggregator','mzXML Aggregator','Extracts mzXML results from multiple mzXML jobs and aggregates the result',0,0,1,0,250,null,'Y','','','mzXML_Aggregator',5,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (34,'Phospho_FDR_Aggregator','PhosphoFDRAggregator','Calculates phosphoproteomics FDR for datasets and aggregates the results',0,0,1,0,250,'','Y','\\gigasax\DMS_Parameter_Files\AScore','','Phospho_FDR_Aggregator',6,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (35,'MSGF','Peptide Extraction','Runs MSGF, aka the MS-GeneratingFunction (CPU Load is 4 to keep to just one or two tasks per box)',0,0,2,0,2048,'','Y','','','MSGF',0.25,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (36,'MSXML_Bruker','XML File Generator','Creates mzXML or mzML files from Bruker .D folders using CompassXtract',1,0,1,0,250,'<section name="MSXMLGenerator"><item key="MSXMLGenerator" value="CompassXport.exe" /><item key="MSXMLOutputType" value="mzXML" /><item key="CentroidMSXML" value="True" /></section>','Y','\\gigasax\DMS_Parameter_Files\MSXML_Bruker','','MSXML_Bruker',13,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (37,'MultiAlign','Peak matching','Matches LCMSFeatureFinder results across datasets and/or to an AMT tag database',0,0,1,0,250,null,'Y','\\gigasax\DMS_Parameter_Files\MultiAlign','','MultiAlign',10,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (38,'MultiAlign_Aggregator','Peak Matching','Matches LCMSFeatureFinder results across datasets and/or to an AMT tag database',0,0,1,0,250,null,'Y','','','MultiAlign_Aggregator',10,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (39,'MSGFPlus','Peptide Search','Runs MSGF+ (CPU load is 4 to allow for debugging, but MSGF+ will use 1 fewer than the number of cores)',0,0,5,0,4096,null,'N','\\gigasax\DMS_Parameter_Files\MSGFPlus','Actually uses all available cores; CPU_Load set to 5 to limit to our newer machines; uses 75% of core count on Proto storage servers','MSGFPlus',300,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (40,'MSDeconv','Deisotoping','Deisotopes peptides with charge 4+ or higher; intended for top-down or middle-down datasets',0,0,1,0,2048,null,'Y','','','MSDeconv',30,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (41,'MSAlign','Peptide Search','Searches MSAlign result files for peptides',0,0,7,1,16000,null,'N','\\gigasax\DMS_Parameter_Files\MSAlign','','MSAlign',1440,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (42,'Mage','Data Extraction','Performs various types of data extraction',0,0,1,0,250,null,'Y','','','Mage',1.5,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (43,'APE','Data Filtering and Aggregation','Runs workflows to filter and aggregate data',0,0,1,0,250,null,'Y','','','APE',0.75,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (44,'Cyclops','Statistical Analysis','Performs various statistical analysis',0,0,1,0,250,null,'Y','','','Cyclops',0.75,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (45,'SMAQC','SMAQC','Runs SMAQC.  Also ran LLRC from June 2013 to November 2015, but disabled in November 2015 because requires R 2.x and we have updated to R 3.x',0,0,1,0,250,null,'Y','\\gigasax\DMS_Parameter_Files\SMAQC','','SMAQC',0.4,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (46,'IDPicker','Protein Grouping','Previously ran IDPicker (disabled in January 2015).  Creates a .pepXML file using PHRP results',0,0,1,0,250,null,'Y','\\gigasax\DMS_Parameter_Files\IDPicker','','IDPicker',0.5,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (47,'IDM','ITRAQ Interference Detection','Tool for calculating interference within isolation window',0,0,1,0,250,'','Y','','','IDM',15,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (48,'AScore','Standard Processing for DMS tool','Calculates phosphoproteomics FDR for datasets and aggregates the results',0,0,1,0,250,'','Y','\\gigasax\DMS_Parameter_Files\AScore','','AScore',0.25,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (49,'LipidMapSearch','Lipid Search','Searches a single dataset or a pair of datasets (positive mode and negative mode) against the Lipid Maps database',0,0,1,0,250,null,'Y','\\gigasax\DMS_Parameter_Files\LipidMapSearch','','LipidMapSearch',5,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (50,'MSAlign_Quant','Peptide Extraction','Quantifies MSAlign results',0,0,1,0,1024,null,'N','\\gigasax\DMS_Parameter_Files\DeconToolsWorkflows','','MSAlign_Quant',1440,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (51,'IDP_Mac','Protein Grouping','Runs IDPicker for the MAC pipline (as of September 2013, this tool is not used by any scripts)',0,0,1,0,250,null,'Y','\\gigasax\DMS_Parameter_Files\IDP_MAC','','IDP_MAC',0.5,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (52,'DataImport','Data Import','Copies data files from an external source into an analysis job folder',0,0,1,0,64,null,'Y','','','DataImport',5,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (53,'MSGFPlus_IMS','Peptide Search','Processes IMS MSn data with MSGFDB_IMS; requires a DeconTools job as import',0,0,5,0,4096,null,'N','\\gigasax\DMS_Parameter_Files\MSGFPlus','Actually uses all available cores; CPU_Load set to 5 to limit to our newer machines','MSGFPlus_IMS',300,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (54,'ProSight_Quant','Peptide Quantitation','Quantifies ProSightPC results',0,0,1,0,1024,null,'N','\\gigasax\DMS_Parameter_Files\DeconToolsWorkflows','','ProSight_Quant',3,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (55,'PRIDE_Converter','PSM Result Aggregator','Converts Peptide_Hit results (Sequest, X!Tandem, or MSGFDB) to the msgf-pride.xml format',0,0,1,0,250,null,'Y','\\gigasax\DMS_Parameter_Files\PRIDE_Converter','','PRIDE_Converter',30,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (56,'MSAlign_Histone','Peptide Search','Searches MSAlign result files for histone peptides',0,0,7,1,16000,null,'N','\\gigasax\DMS_Parameter_Files\MSAlign_Histone','','MSAlign_Histone',1440,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (57,'Results_Cleanup','Utility','For MAC jobs, looks for Results.db3 files and deletes all except the one in the final job step directory.  For MaxQuant jobs, deletes files from the transfer directory that we don''t want to keep long term.  Specifically, deletes the combined directory, Dataset.index files, and each dataset''s .zip file',0,0,1,0,250,null,'Y','','','Results_Cleanup',0.2,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (58,'RepoPkgr','PSM Result Aggregator','Obsolete (script PeptideAtlas was retired in 2020): Aggregates data and results files from DMS into package that can be uploaded to a public proteomics repository',0,0,1,0,250,null,'Y',null,'',null,1,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (59,'MSGFPlus_HPC','Peptide Search','Runs MSGF+ on the Deception HPC Compute Cluster (server admin: Clay Hagler)',0,0,1,0,250,null,'N','\\gigasax\DMS_Parameter_Files\MSGFPlus','','MSGFPlus_HPC',300,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (60,'MODa','Peptide Search','Runs MODa',0,0,7,1,3000,null,'N','\\gigasax\DMS_Parameter_Files\MODa','','MODa',1440,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (61,'DeconPeakDetector','Data Extraction','Detects peaks in mass spectra using the DeconTools peak finder or the Hammer peak finder; used by GlyQ-IQ',0,0,1,0,250,null,'Y','\\gigasax\DMS_Parameter_Files\DeconPeakDetector','','DeconPeakDetector',15,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (62,'GlyQ_IQ','Glycomics Search','Searches a dataset for signatures of glycans using GlyQ-IQ',0,0,7,1,4000,null,'Y','\\gigasax\DMS_Parameter_Files\GlyQ-IQ','','GlyQ_IQ',180,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (63,'PBF_Gen','PBF Generator','Creates PBF (PNNL Binary Format) files; if a new .Exe results in a different file format, then increment PbfFormatVersion in MSPF_TopDown_Standard.xml and possibly increment Shared_Result_Version in this table',1,0,1,0,250,'<section name="PbfGenerator"><item key="PbfGenVersion" value="v1.0.5300" /></section>','Y','(na)','','PBF_Gen',12,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (64,'MSPathFinder','Peptide or protein Search','Searches top down datasets for proteins',0,0,4,0,8000,null,'Y','\\gigasax\DMS_Parameter_Files\MSPathFinder','','MSPathFinder',600,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (65,'Mz_Refinery','XML File Generator','Updates the m/z values in a .mzML file based on the results from a MSGF+ search',1,0,5,0,4000,'<section name="ScanControl"><item key="ScanStart" value="1" /><item key="ScanStop" value="999999" /><item key="MaxIntermediateScansWhenGrouping" value="1" /></section><section name="MWControl"><item key="MWStart" value="200" /><item key="MWStop" value="5000" /></section><section name="IonCounts"><item key="IonCount" value="35" /></section><section name="MassTol"><item key="MassTol" value="3" /></section><section name="DtaGenerator"><item key="DtaGenerator" value="DeconMSN.exe" /></section><section name="Charges"><item key="CreateDefaultCharges" value="True" /><item key="ExplicitChargeStart" value="0" /><item key="ExplicitChargeEnd" value="0" /></section><section name="MSXMLGenerator"><item key="MSXMLGenerator" value="MSConvert.exe" /><item key="MSXMLOutputType" value="mzML" /><item key="CentroidMSXML" value="True" /><item key="CentroidPeakCountToRetain" value="-1" /><item key="RecalculatePrecursors" value="False" /></section><section name="MzRefinery"><item key="MzRefParamFile" value="MzMLRef_NoMods.txt" /></section>','Y','\\gigasax\dms_parameter_Files\MzRefinery','CPU_Load set to 5 to match MSGF+','Mz_Refinery',25,1,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (66,'ProMex','Deisotoping','Deisotopes high resolution spectra in PBF files',1,0,4,0,2048,'<section name="PbfGenerator"><item key="PbfFormatVersion" value="150605" /></section><section name="ProMex"><item key="ProMexParamFile" value="ParamFileName.txt" /><item key="BinResolutionPPM" value="16" /></section>','Y','\\gigasax\dms_parameter_Files\ProMex','','ProMex',60,1,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (67,'Bruker_DA_Export','XML File Generator','Creates XML files from Bruker .D folders',0,0,1,0,1024,null,'Y','\\gigasax\DMS_Parameter_Files\Bruker_Data_Analysis','','Bruker_DA',1.25,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (68,'NOMSI','Peptide Search','Process XML files created by Bruker_DA_Export',0,0,1,0,1024,null,'Y','\\gigasax\DMS_Parameter_Files\NOMSI','','NOMSI',0.1,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (69,'MODPlus','Peptide Search','Runs MODPlus',0,0,7,1,9000,null,'Y','\\gigasax\DMS_Parameter_Files\MODPlus','','MODPlus',1440,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (70,'QC_ART','QCART','Runs QC-ART; requires R 3.x. CPU_Load is 0 because virtually no system resources are utilized',0,0,0,0,250,null,'Y','\\gigasax\DMS_Parameter_Files\QC-ART','','QCA',0.2,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (71,'Test_MSGFPlus','Peptide Search','Runs MSGF+',0,0,5,0,4096,null,'N','\\gigasax\DMS_Parameter_Files\MSGFPlus','Actually uses all available cores; CPU_Load set to 5 to limit to our newer machines; uses 75% of core count on Proto storage servers','MSGFPlus',300,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (72,'DataExtractorSplitFasta','Peptide Extraction','Creates loadable peptide ID file and first hist file; used by script MSGFPlus_SplitFasta to assure this tool runs on machines with sufficient memory; will also run AScore if we searched for Phospho STY',0,0,2,0,4000,null,'Y','','The MzidMerger tool will use several GB of memory when a large FASTA file was used','DataExtractor',5,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (73,'Formularity','Formula Search','Process XML files created by Bruker_DA_Export',0,0,1,0,1024,null,'Y','\\gigasax\dms_parameter_Files\Formularity','','Formularity',1,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (74,'TopFD','Deisotoping','Deisotopes peptides; intended for top-down or middle-down datasets',0,0,7,0,750,null,'Y','\\gigasax\DMS_Parameter_Files\TopFD','','TopFD',30,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (75,'TopPIC','Peptide Search','Searches _ms2.msalign files for peptides',0,0,7,0,2048,null,'Y','\\gigasax\DMS_Parameter_Files\TopPIC','','TopPIC',600,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (76,'ThermoPeakDataExporter','Data Extraction','Exports peak data (m/z and intensity) from a Thermo raw file to create a tab-delimited text file',0,0,1,0,1024,null,'Y','','','ThermoPeakData',1,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (77,'MSFragger','Peptide Search','Searches .mzML files for peptides, using either a traditional closed search or an open search (which finds unknown dynamic mods)',0,0,7,1,4000,null,'Y','\\gigasax\DMS_Parameter_Files\MSFragger','','MSFragger',60,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (78,'PepProtProphet','Protein Grouping','Runs peptide prophet and protein prophet using Philosopher',0,0,1,0,2048,null,'Y','','','PepProtProphet',15,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (79,'MaxqPeak','Peptide Search','Searches .raw or .mzML files for peptides',0,0,9,1,8000,null,'Y','\\gigasax\DMS_Parameter_Files\MaxQuant','','MaxQuant',120,0,1,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (80,'MaxqS1','Peptide Search','Runs after a MaxqPeak step when a MaxQuant search is split into segments',0,0,9,1,8000,null,'Y','\\gigasax\DMS_Parameter_Files\MaxQuant','','MaxQuant',120,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (81,'MaxqS2','Peptide Search','Runs after a MaxqPeak step when a MaxQuant search is split into segments',0,0,9,1,8000,null,'Y','\\gigasax\DMS_Parameter_Files\MaxQuant','','MaxQuant',120,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (82,'MaxqS3','Peptide Search','Runs after a MaxqPeak step when a MaxQuant search is split into segments',0,0,9,1,8000,null,'Y','\\gigasax\DMS_Parameter_Files\MaxQuant','','MaxQuant',120,0,0,2)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (83,'DIA-NN_SpecLib','Spectral Library Generator','Create an in-silico digest based spectrum library using a FASTA file',0,0,7,1,8000,null,'Y','\\gigasax\DMS_Parameter_Files\DIA-NN','','DIA-NN_SpecLib',60,0,0,10)
INSERT INTO [T_Step_Tools] (ID, Name, Type, Description, Shared_Result_Version, Filter_Version, CPU_Load, Uses_All_Cores, Memory_Usage_MB, Parameter_Template, Available_For_General_Processing, Param_File_Storage_Path, Comment, Tag, AvgRuntime_Minutes, Disable_Output_Folder_Name_Override_on_Skip, Primary_Step_Tool, Holdoff_Interval_Minutes) VALUES (84,'DIA-NN','Peptide Search','Runs DIA-NN to identify peptides in DIA spectra using a spectrum library',0,0,7,1,8000,null,'N','\\gigasax\DMS_Parameter_Files\DIA-NN','','DIA-NN',30,0,1,2)
SET IDENTITY_INSERT [T_Step_Tools] OFF
